cve_bin_tool/checkers/firefox.py

```diff
@@ -0,0 +1,23 @@
+# Copyright (C) 2022 Orange
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+
+"""
+CVE checker for firefox
+
+https://www.cvedetails.com/product/3264/Mozilla-Firefox.html?vendor_id=452
+
+"""
+from __future__ import annotations
+
+from cve_bin_tool.checkers import Checker
+
+
+class FirefoxChecker(Checker):
+    CONTAINS_PATTERNS: list[str] = []
+    FILENAME_PATTERNS: list[str] = []
+    VERSION_PATTERNS = [
+        r"firefox\-[0-9][0-9]([0-9]?)+\.[0-9]+\.[0-9]+",
+        r"firefox\-l10n+\-[a-z][a-z]+\_[0-9][0-9]([0-9]?)+\.[0-9]+\-[0-9]+",
```

Comment:
The tests are failing since the patterns here aren't detected in the parsed strings from the binaries(condensed files). Also the test string provided (line 8) won't be matched either.

How did you find the patterns? Using helper script or did you follow the checker readme?

Reply:
I initially followed the checker readme but could not get any fruitful output, so I modelled it bases on previously present checkers.

Reply:
Can you run the helper_script over any of the firefox package? From what I tried there are some viable patterns inside the binary.

Refer https://cve-bin-tool.readthedocs.io/en/latest/CHECKERS.html?highlight=helper#helper-script to see how to use it.

Reply:
Also the sqlite error happened because there's no existing CVE database in the cache for cve-bin-tool to use.
You can run something like `python -m cve_bin_tool.cli test test/condensed-downloads/dnsmasq-2.79-13.el8_3.1.aarch64.rpm.tar.gz` to initialise a working database

Reply:
This is coming on running `python -m cve_bin_tool.cli test test/condensed-downloads/dnsmasq-2.79-13.el8_3.1.aarch64.rpm.tar.gz`
![Screenshot from 2022-11-07 02-04-08](https://user-images.githubusercontent.com/98874428/200193730-b40633c7-7dff-4463-aff4-8878067f9c06.png)

Reply:
I'm not sure why that's happening but there's a mistake in the sample command I shared with you. Try `python -m cve_bin_tool.cli test/condensed-downloads/dnsmasq-2.79-13.el8_3.1.aarch64.rpm.tar.gz`.
If that didn't work try running `python setup.py egg_info` to load checkers properly and run the other command again.

Reply:
Still not happening

![Screenshot from 2022-11-07 10-59-17](https://user-images.githubusercontent.com/98874428/200233127-8845ac32-1fc9-4b74-9ec9-f92968acacca.png)

Reply:
The error seems to be thrown from `reportlab` which is an optional dependency of cve-bin-tool. Usually it shouldn't be installed if you installed the dependencies from `requirements.txt`. Can you try a fresh install and setup the dev environment following https://cve-bin-tool.readthedocs.io/en/latest/CONTRIBUTING.html? Your commited and pushed changes can still be accessed in the newer copy.

Reply:
Done it but still same issue. Also while trying to seperately install reportlab, it shows 'requirements already met'.

Reply:
Done. Updating reportlab did the trick.

Reply:
@BreadGenie

Reply:
the tests are failing because there's no string that matches the regex patterns in the `VERSION_PATTERNS`. Here it's due to it being an empty list.

What you gave to do is find is a string in the binary that have a unique pattern with version and possibly the product name using helper script or the readme in the checkers folder.

Reply:
In the helper script it is coming empty strings. I have copied it as it is from the helper script.

Reply:
There could be some binaries without any strings that aren't detected by the helper script. You can try either more binaries or follow the manual method in the checker readme.

I have went ahead and tried it on couple binaries and found 2 strings that are interesting.
`r"ENCODER=Mozilla106.0(?:(?:\r?\n.*?)*)CN=Firefox U2F Soft Token"` and
`r"{"type":"rpm","name":"firefox","version":"106.0.3-1.fc38","architecture":"x86_64","osCpe":"cpe:/o:fedoraproject:fedora:38"}"`

For the 1st string we can create a pattern like `r"ENCODER=Mozilla([0-9]+\.[0-9]+(\.[0-9]+)?).*Firefox"

Here we will try to match strings that starts with `ENCODER=Mozilla` and ends with `Firefox` (added it for less chance of false positives) with the version capturing group in the middle. The version capturing group can match versions like 106.0 and 106.1.1 etc. So if the test string is `ENCODER=Mozilla106.0(?:(?:\r?\n.*?)*)CN=Firefox U2F Soft Token"` it'll match and captures the version as 106.0.

I'll leave rest of the exploration to you ;-)